### PR TITLE
Update outdated testing documentation (use pytest)

### PR DESCRIPTION
**Pull request by SubaruArai**
The CONTRIBUTING.md's "How to submit pull requests" specified nosetests, which is no longer used. (see: commit id f27dfbf, github pr #36201)
This commit updates the documentation to use pytest instead, and also encourages the use of a virtual environment.

Config files are also updated to ignore the virtual environment.

### CODE REVIEW

**Review by clalancette — CHANGES_REQUESTED** [review]
I'll note that this will conflict with https://github.com/ros/rosdistro/pull/38125 .  There is more information in here than in that PR, but I think we should probably get that one in first and then rebase this on top.

**Comment by SubaruArai** [comment]
@clalancette merged, please check!

**Comment by SubaruArai** [comment]
@clalancette ping

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -318,10 +337,17 @@ These tests require several dependencies that can be installed either from the R
 | unidiff        | python3-unidiff (Zesty and higher) | unidiff        |
 | yamllint       | yamllint                          | yamllint       |
```
Comment (clalancette) [inline CONTRIBUTING.md:338]: This whole thing seems a bit redundant.  We tell the user 3 times to run `pytest` locally.

What I'm going to suggest instead is that we rewrite this to be simpler.  In particular, let's just add in one section called "Unit testing" or something like that.  In there, we say something like:

```
All of the dependencies for testing are listed in tests/requirements.txt.  It is recommended to use a virtual environment and pip to install the dependencies.
```

And then go on to give those instructions on how to install and run.  We can then remove the rest of the duplicated information here, so this is a lot shorter and more to the point.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -318,10 +337,17 @@ These tests require several dependencies that can be installed either from the R
...
+```bash
+for f in ./rosdep/*.yaml; do echo "$f"; ./scripts/check_rosdep.py "$f"; done
+```
 
-For example, to check a change to `rosdep/base.yaml`:
+Or simply use `pytest` (although the output is quite verbose):
 ```bash
-python3 scripts/check_rosdep.py rosdep/base.yaml
+pytest test/rosdep_formatting_test.py
 ```
```
Comment (clalancette) [inline CONTRIBUTING.md:353]: Honestly, I'm not sure we need this bit at all.  This is already covered by the general testing above, so maybe we can just drop this stanza completely?

**Review by clalancette — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (SubaruArai) [reply CONTRIBUTING.md:338]: lgtm

File: CONTRIBUTING.md (hunk 2 above)
Comment (SubaruArai) [reply CONTRIBUTING.md:353]: Yeah, this was basically a leftover that I didn't know if I should've deleted.
I'm fine for dropping.

**Comment by SubaruArai** [comment]
@clalancette updated, please check!

**Comment by SubaruArai** [comment]
@clalancette ping

**Comment by SubaruArai** [comment]
@clalancette ping

**Review by clalancette — APPROVED** [review]
Sorry for the delay here.  This looks good to me now.

**Pull request merged**